guardian/grid
# Conal/imagedam 1502 send to capture handle user input

## What does this change?

As an archivist user of BBC Images, I want to be able to send certain images from BBC Images to BBC Photo Sales, through the Grid UI. This PR introduces this functionality, allowing a user to select a number of images and trigger the outbound photo sales process by clicking on the new 'Send to Photo Sales' button. By doing this, a syndication usage (with a 'pending' usage status) is added to each of the selected image, which is the trigger for an AWS step function that carries out the majority of the outbound logic.  

This PR is only concerned with the ability of a user to select images and add a syndication usage through the frontend, future PRs will handle additional features such as confirmation dialogs, restrictions on what images can be sent, success/failure notifications and changes to the metadata panel.

As this is BBC specific functionality, it is disabled by default via the `showSendToPhotoSales` feature flag.

![Screenshot 2024-03-20 at 16 56 07](https://github.com/guardian/grid/assets/129299738/eb16a5ff-550f-4857-9557-c67fad33266e)

Note: this PR has been developed on top of the changes introduced in https://github.com/guardian/grid/pull/4244, which is still to be merged. As such there are additional files inlcuded in the PR that are outside its scope. The only files that need to be reviewed as part of this PR are:

- kahuna/app/lib/KahunaConfig.scala
- kahuna/app/views/main.scala.html
- kahuna/public/js/search/results.html
- kahuna/public/js/search/results.js
- kahuna/public/js/services/api/media-api.js
- media-api/app/controllers/MediaApi.scala
- media-api/conf/routes

## How should a reviewer test this change?

As a user with elevated permissions and with the `showSendToPhotoSales` flag set to true:

- Select a number of image thumbnails from the main grid view
- Click the 'Send to Photo Sales Button' from the menu that appears

## How can success be measured?
- Confirm that a Pending Publication usage is visible in the usages section of the metadata panel, for the selected images

## Who should look at this?
@guardian/digital-cms

## Tested? Documented?
